### ISSUE TITLE
gas required exceeds allowance (12487735) or always failing transaction

### ISSUE THREAD

**Issue by sandybradley** [brownie-mix/aave-flashloan-mix#4]
Hi, Thank you for this repo. Very nice. 

I deployed a contract, on the mainnet, based on your kit with my own parts added. When I try to call the flashloan function I get the following error. 

Any advice ?

>>> acct
<LocalAccount '0x513E95Eec3bAC74802b4b2D859e802f17eAE57a9'>
>>> flashloan
<MyFlashloanContract Contract '0x80be9fBea39Ae56d090148c4EF96f7f7f4Dc9Aa1'>
>>> tx = flashloan.flashloan(eth_reserve,10000000000000000,chai_reserve, {"from": acct})
  File "<console>", line 1, in <module>
  File "c:\users\alexander, line line, in in
    return self.transact(*args)
  File "c:\users\alexander, line line, in in
    return tx["from"].transfer(
  File "c:\users\alexander, line line, in in
    "gas": Wei(gas_limit) or self._gas_limit(to, amount, gas_buffer, data),
  File "c:\users\alexander, line line, in in
    gas_limit = self.estimate_gas(to, amount, data or "")
  File "c:\users\alexander, line line, in in
    return web3.eth.estimateGas(
  File "c:\users\alexander, line line, in in
    return self.web3.manager.request_blocking(
  File "c:\users\alexander, line line, in in
    raise ValueError(response["error"])
ValueError: {'code': -32000, 'message': 'gas required exceeds allowance (12463323) or always failing transaction'}
>>>

Best,
Alexander

**Comment by iamdefinitelyahuman** [comment]
That's a common error message thrown by a node when your transaction will revert.  Hard to say why it's reverting without more context.

In cases like this I usually test using Ganache's `--fork` mode, where the full range of debug functionality is available and a mistake doesn't cost me any money.  This article might help - https://medium.com/@iamdefinitelyahuman/ethereum-mainnet-testing-with-python-and-brownie-82a61dee0222

**Comment by sandybradley** [comment]
Thank you for your prompt response, Ben. 

I did read through all the documentation and tested most functionality on the mainnet fork. My contract is an experimental simple arbitrageur, based on the example flashloan from aave (v0.5.0) and using 1split (https://etherscan.io/address/1split.eth) for 2 swaps. The only thing I could not get to work on the fork mode were the swaps themselves. I figure the 1split contract would only work on the mainnet so gave it a go. The only arb opportunity to test it on yesterday was eth-chai , which was what produced the error in the previous message. I tried to do the same Eth-Chai trades on Furucombo (https://furucombo.app/combo),  and the transactions also failed for unknown reasons. 

I'll do some more testing and let you know how I get on. Thanks again.
